app/src/ui/preferences/git.tsx

```diff
@@ -29,7 +54,82 @@ export class Git extends React.Component<IGitProps, {}> {
             onValueChanged={this.props.onEmailChanged}
           />
         </Row>
+
+        {this.renderDefaultBranchSetting()}
       </DialogContent>
     )
   }
+
+  private renderWarningMessage = (
+    sanitizedBranchName: string,
+    proposedBranchName: string
+  ) => {
+    if (sanitizedBranchName === '') {
+      return (
+        <>
+          <Ref>{proposedBranchName}</Ref> is an invalid branch name.
+        </>
+      )
+    }
+
+    return (
+      <>
+        Will be saved as <Ref>{sanitizedBranchName}</Ref>.
+      </>
+    )
+  }
+
+  private renderDefaultBranchSetting() {
+    if (!enableDefaultBranchSetting()) {
+      return null
+    }
+
+    const defaultBranchIsOther = !SuggestedBranchNames.includes(
+      this.props.defaultBranch
+    )
+
+    return (
+      <>
+        <p className="default-branch-title">
+          Default branch on new repositories
+        </p>
```

Comment:
We use `h2` in the advanced tab for section titles, let's mirror that there and I'd suggest we say "Default branch **for** new..." instead of **on**

```suggestion
        <h2>
          Default branch for new repositories
        </h2>
```